python/cudf/cudf/_lib/pylibcudf/scalar.pxd

```diff
@@ -0,0 +1,32 @@
+# Copyright (c) 2023, NVIDIA CORPORATION.
+
+from libcpp cimport bool
+from libcpp.memory cimport unique_ptr
+
+from rmm._lib.memory_resource cimport DeviceMemoryResource
+
+from cudf._lib.cpp.scalar.scalar cimport scalar
+
+from .types cimport DataType
+
+
+cdef class Scalar:
+    cdef unique_ptr[scalar] c_obj
+    cdef DataType _data_type
+
+    # Holds a reference to the DeviceMemoryResource used for allocation.
+    # Ensures the MR does not get destroyed before this DeviceBuffer. `mr` is
+    # needed for deallocation
+    cdef DeviceMemoryResource mr
+
+    cdef const scalar* get(self) except *
+
+    cpdef DataType type(self)
+    cpdef bool is_valid(self)
+
+    @staticmethod
+    cdef Scalar from_libcudf(unique_ptr[scalar] libcudf_scalar, dtype=*)
+
+    # TODO: Make sure I'm correct to avoid typing the metadata as
```

Comment:
Should we resolve before merging?

Reply:
Yup. This comment is correct and can be removed. Depending on the resolution of your other comment though (combining the Scalar method and the interop function) this may not be an issue, let me see.

Reply:
OK, with the changes to make the class methods the primary means of going between arrow and pylibcudf Scalar/Table objects this limitation is no longer present. I've typed the method appropriately.